packages/amplify-cli/src/lib/attach-backend.js

```diff
@@ -0,0 +1,120 @@
+const fs = require('fs-extra');
+const path = require('path');
+const util = require('util');
+const queryProvider = require('./attach-backend-steps/a10-queryProvider');
+const analyzeProject = require('./attach-backend-steps/a20-analyzeProject');
+const initFrontend = require('./attach-backend-steps/a30-initFrontend');
+const generateFiles = require('./attach-backend-steps/a40-generateFiles');
+const { constructInputParams, postPullCodeGenCheck } = require('./amplify-service-helper');
+
+const backupAmplifyDirName = 'amplify-backup';
+
+async function attachBackend(context) {
+  backupAmplifyFolder(context);
+  setupFolderStructure(context);
+  prepareContext(context);
+  await queryProvider
+    .run(context)
+    .then(analyzeProject.run)
+    .then(initFrontend.run)
+    .then(generateFiles.run)
+    .then(onSuccess)
+    .catch(e => {
+      removeFolderStructure(context);
+      restoreOriginalAmplifyFolder(context);
+      context.print.error('Failed to pull the backend.');
+      context.print.info(util.inspect(e));
+      process.exit(1);
+    });
+}
+
+async function onSuccess(context) {
+  const { inputParams } = context.exeInfo;
+  if (inputParams.amplify.noOverride) {
+    const projectPath = process.cwd();
+    const backupAmplifyDirPath = path.join(projectPath, backupAmplifyDirName);
+    const backupBackendDirPath = path.join(backupAmplifyDirPath, context.amplify.constants.BackendamplifyCLISubDirName);
+    if (fs.existsSync(backupBackendDirPath)) {
+      const backendDirPath = context.amplify.pathManager.getBackendDirPath(projectPath);
+      fs.removeSync(backendDirPath);
+      fs.copySync(backupBackendDirPath, backendDirPath);
+    }
+  }
+
+  await postPullCodeGenCheck(context);
+
+  if (!inputParams.yes) {
+    const confirmKeepCodebase = await context.amplify.confirmPrompt.run('Do you want to keep the backend codebase?', true);
+    if (confirmKeepCodebase) {
+      context.print.info('');
+      context.print.success('Backend environment has been successfully pulled and attached to your project.');
```

Comment:
wording: what do you think about "the" instead of "your"?

Reply:
wording is decided by the PM.

Reply:
changed